lib/improver/blending/weighted_blend.py

```diff
@@ -454,165 +454,216 @@ def process(self, cube, weights=None):
                    'iris.cube.Cube but is'
                    ' {0:s}.'.format(type(cube)))
             raise TypeError(msg)
 
-        # Check that the points within the time coordinate are equal
-        # if the coordinate being blended is forecast_reference_time.
-        if self.coord == "forecast_reference_time":
-            if cube.coords("time"):
-                time_points = cube.coord("time").points
-                if len(np.unique(time_points)) > 1:
-                    msg = ("For blending using the forecast_reference_time "
-                           "coordinate, the points within the time coordinate "
-                           "need to be the same. The time points within the "
-                           "input cube are {}".format(time_points))
-                    raise ValueError(msg)
-
-        # Check to see if the data is percentile data
+    @staticmethod
+    def _check_all_time_points_equal(cube):
+        if cube.coords("time"):
+            time_points = cube.coord("time").points
+            if len(np.unique(time_points)) > 1:
+                msg = ("For blending using the forecast_reference_time "
+                       "coordinate, the points within the time coordinate "
+                       "need to be the same. The time points within the "
+                       "input cube are {}".format(time_points))
+                raise ValueError(msg)
+
+    def _check_weights_have_coord_shape(self, cube, weights):
+        if np.array(weights).shape != cube.coord(self.coord).points.shape:
+            msg = ('The weights array must match the shape '
+                   'of the coordinate in the input cube; '
+                   'weight shape is '
+                   '{0:s}'.format(np.array(weights).shape) +
+                   ', cube shape is '
+                   '{0:s}'.format(cube.coord(self.coord).points.shape))
+            raise ValueError(msg)
+
+    def _perform_blending(self, cube, weights):
+        cubelist = iris.cube.CubeList([])
+        for cube_thres in self._slices_over_threshold(cube):
+            cube_new = self._perform_blend_across_threshold(cube_thres,
+                                                            weights)
+            cube_new = conform_metadata(
+                cube_new, cube_thres, coord=self.coord,
+                cycletime=self.cycletime,
+                coords_for_bounds_removal=self.coords_for_bounds_removal)
+
+            cubelist.append(cube_new)
+        result = cubelist.merge_cube()
+        if isinstance(cubelist[0].data, np.ma.core.MaskedArray):
+            result.data = np.ma.array(result.data)
+        return result
+
+    def _perform_blend_across_threshold(self, cube_thres, weights):
+
+        # check to see if the data is percentile data
+        # if not, perc_coord will be None
+        perc_coord = self._get_perc_dim_coord(cube_thres)
+
+        # Blend the cube across the coordinate
+        if self.mode == "weighted_mean":
```

Comment:
An argument could be made that the `_blend_weighted_mean_percentile`, `_blend_weighted_mean`, and `_blend_weighted_maximum` should be declared in that order as methods lower down. If so it would match this call order and the diff would be easier! Up to you.

Reply:
You're right - missed that one.